bindings/src/main/java/io/pravega/storage/extendeds3/ExtendedS3StorageFactory.java

```diff
@@ -7,23 +7,25 @@
  *
  *     http://www.apache.org/licenses/LICENSE-2.0
  */
-package io.pravega.segmentstore.storage.impl.extendeds3;
+package io.pravega.storage.extendeds3;
 
 import com.emc.object.s3.S3Config;
 import com.emc.object.s3.jersey.S3JerseyClient;
 import com.google.common.base.Preconditions;
+import io.pravega.common.ConfigSetup;
 import io.pravega.segmentstore.storage.AsyncStorageWrapper;
 import io.pravega.segmentstore.storage.Storage;
 import io.pravega.segmentstore.storage.StorageFactory;
 import io.pravega.segmentstore.storage.rolling.RollingStorage;
 import java.util.concurrent.ExecutorService;
+import java.util.concurrent.ScheduledExecutorService;
 
 /**
  * Factory for ExtendedS3 Storage adapters.
  */
 public class ExtendedS3StorageFactory implements StorageFactory {
-    private final ExtendedS3StorageConfig config;
-    private final ExecutorService executor;
+    private ExtendedS3StorageConfig config;
```

Comment:
If these are non-final they need to be guarded by a lock, and annotated accordingly. (See my comment below on how to avoid this)

Reply:
Fixed.